lib/web_ui/lib/src/engine/window.dart

```diff
@@ -214,17 +217,30 @@ class EngineWindow extends ui.Window {
     scheduleFrameCallback!();
   }
 
+  @visibleForTesting
+  JsUrlStrategy? urlStrategy = ui.debugEmulateFlutterTesterEnvironment
+      ? null
+      : convertToJsUrlStrategy(const HashUrlStrategy());
+
   /// Change the strategy to use for handling browser history location.
   /// Setting this member will automatically update [_browserHistory].
   ///
   /// By setting this to null, the browser history will be disabled.
-  set locationStrategy(LocationStrategy? strategy) {
-    _browserHistory.setLocationStrategy(strategy);
+  Future<void> setUrlStrategy(JsUrlStrategy? strategy) {
+    _isHistoryInitialized = true;
+    urlStrategy = strategy;
+    return _browserHistory.setUrlStrategy(strategy);
```

Comment:
I'm not sure `BrowserHistory` needs an explicit setter for the strategy. It should be able to get it from the top-level `get urlStrategy`.

Reply:
`BrowserHistory` needs to do some processing when the URL strategy changes. I'll see if we can simplify this now that we don't allow "updating" the URL strategy.